webbpsf/webbpsf_core.py

```diff
@@ -2252,13 +2256,22 @@ def _addAdditionalOptics(self, optsys, oversample=2):
             SAM_box_size = 5.0
         elif ((self.image_mask == 'MASKSWB') or (self.image_mask == 'MASKLWB')):
             bar_offset = self.options.get('bar_offset', None)
-            # If the bar offset is not provided, use the filter name to lookup the default
+            # If the bar offset is not provided, use the SIAF aperture name, or else the filter name to lookup the default
             # position. If an offset is provided and is a floating point value, use that
             # directly as the offset. Otherwise assume it's a filter name and try passing
             # that in to the auto offset. (that allows for selecting the narrow position, or
             # for simulating using a given filter at some other filter's position.)
             if bar_offset is None:
-                auto_offset = self.filter
+                # Try to use the SIAF aperture name to determine the offset
```

Comment:
@mperrin I think this is the addition that's creating problems in test_return_intermediates. In the test,  auto_offset = maskswb because aperture name, 'NRCA4_FULL_MASKSWB' and that's not an option in offset_swb. In the cases of maskswb/masklwb, it should  be setting auto_offset = self.filter as before but I guess this breaks something for some observations when matching observations.  We could change the test by assigning the offset manually:

`filt = 'F182M'
nc = webbpsf_core.NIRCam()
nc.filter = filt
nc.image_mask='maskswb'
nc.pupil_mask='wedgelyot'
coron = webbpsf.optics.NIRCam_BandLimitedCoron('MASKSWB')
bar_offset = coron.get_bar_offset_from_siaf(filt, channel='SW')
nc.options['bar_offset']  = bar_offset

osys = nc.get_optical_system()

psf, intermediates = nc.calc_psf(monochromatic=2e-6, return_intermediates=True)
assert len(intermediates) == len(osys.planes)
assert isinstance(intermediates[0], poppy.Wavefront)
assert isinstance(psf, astropy.io.fits.HDUList)`

Reply:
I fixed the code to eliminate this bug. This test passes now. 

Changing the test code itself is rarely the right way to go, not as a first step. The starting assumption should be that a failing test is telling you there's some kind of bug or logic issue in the code. In the case, the test failure _correctly_ caught that the set of code I'd written here had not considered that sort of input. I wrote it expecting aperture names like "NRCA4_FULL_MASKSWB_F210M" but neglected the case of it being like "NRCA4_FULL_MASKSWB".